hazelcast/src/main/java/com/hazelcast/datalink/HazelcastDataLink.java

```diff
@@ -77,25 +86,19 @@ public void shutdown() {
     }
 
     private ClientConfig buildClientConfig() {
-        ClientConfig clientConfig = null;
-        String clientXml = getConfig().getProperty(CLIENT_XML);
-        if (!StringUtil.isNullOrEmpty(clientXml)) {
-            // Read ClientConfig from XML
-            clientConfig = asClientConfig(clientXml);
-        }
+        validateConfiguration();
 
-        String clientYaml = getConfig().getProperty(CLIENT_YML);
-        if (!StringUtil.isNullOrEmpty(clientYaml)) {
-            // Read ClientConfig from Yaml
-            clientConfig = asClientConfigFromYaml(clientYaml);
-        }
+        HazelcastDataLinkConfigLoader configLoader = new HazelcastDataLinkConfigLoader();
+        DataLinkConfig dataLinkConfig = configLoader.loadConfigFromFile(getConfig());
```

Comment:
I'd change loadConfigFromFile to just `load`, because:
1. It's not necessarily using I/O and that's the first thing which comes into mind when I see "from file"
2. I think class and variable names are descriptive enough as they are

Reply:
rev : e5a7a7c91d81a0833662b44490af811399248446